src/java.base/linux/classes/jdk/internal/platform/CgroupSubsystemFactory.java

```diff
@@ -108,12 +108,12 @@ static CgroupMetrics create() {
         Map<String, CgroupInfo> infos = result.getInfos();
         if (result.isCgroupV2()) {
             // For unified it doesn't matter which controller we pick.
-            CgroupInfo anyController = infos.get(MEMORY_CTRL);
+            CgroupInfo anyController = infos.values().iterator().next();
             CgroupSubsystem subsystem = CgroupV2Subsystem.getInstance(anyController);
-            return subsystem != null ? new CgroupMetrics(subsystem) : null;
+            return new CgroupMetrics(subsystem);
```

Comment:
Should add `Objects.requireNonNull(anyController)` and `Objects.requireNonNull(subsystem)`.

Reply:
I added the first, but not the second as that looks like an overkill; see the definition of the constructor:
```Java
    CgroupMetrics(CgroupSubsystem subsystem) {
        this.subsystem = Objects.requireNonNull(subsystem);
    }
```